### PR TITLE
Add app statuses logic and component

### LINKED ISSUE #6746: Surface provider status in the app

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR resolves https://github.com/dust-tt/dust/issues/6746. For more context see [thread](https://dust4ai.slack.com/archives/C05B529FHV1/p1723139777736699).
Runbook is [here](https://www.notion.so/dust-tt/Runbook-How-to-Update-Provider-Status-Page-566335635569422893fc2bc71e6857e7).

This PR introduces a new API endpoint to retrieve app statuses. Initially, these app statuses will only include provider statuses, but future enhancements will expand this to include more. This naive approach, addresses the following needs:
1. We do not want to update our status page for provider incidents since these incidents are beyond our control.
2. We need a method so that **anyone in the team** can easily surface provider issues within the app.

To meet these requirements, this implementation uses a dedicated status page that exclusively tracks our provider incidents. This status page is public but unlisted, meaning it can only be accessed via a direct link. Due to the aggressive rate limits of the Atlassian status page API, we cache the app status, which is acceptable even if it takes a few minutes for an incident to appear.

When an incident occurs, a banner will appear in the navigation bar, displaying the incident's title, description, and a link to the incident. Currently, this implementation does not support provider-specific logic and can only handle one provider incident at a time.

In the future, we may leverage status page components to represent each of our providers and perform advanced actions in the UI, but this is beyond the scope of the initial implementation.

image

## Deploy Plan

We need to adds the following environment variables in the `front-secrets` secret:
- [x] `PROVIDER_STATUS_PAGE_ID`
- [x] `STATUS_PAGE_API_TOKEN`

### CODE REVIEW

File: front/lib/api/config.ts [hunk 1]
```diff
@@ -121,6 +121,13 @@ const config = {
   getTextExtractionUrl: (): string => {
     return EnvironmentConfig.getEnvVariable("TEXT_EXTRACTION_URL");
   },
+  // Status page.
+  getProviderStatusPageId: (): string => {
+    return EnvironmentConfig.getEnvVariable("PROVIDER_STATUS_PAGE_ID");
+  },
+  getStatusPageApiToken: (): string => {
```
Comment (fontanierh) [inline front/lib/api/config.ts:128]: is that key only for the provider status page ? or works also for main status page ?

**Review by fontanierh — APPROVED** [review]
Looks great ! I was going to argue that we didn't really need a status page for this, but code turns out to be pretty simple so LGTM 👍

File: front/lib/api/config.ts (hunk 1 above)
Comment (flvndvd) [reply front/lib/api/config.ts:128]: It works for both.

**Comment by flvndvd** [comment]
> Looks great ! I was going to argue that we didn't really need a status page for this, but code turns out to be pretty simple so LGTM 👍

This is the whole point of this implementation 🙃.

**Pull request merged**